src/zip.ts

```diff
@@ -151,3 +156,31 @@ export function zip64ExtraField(file: ZipFileDescription, offset: bigint) {
   header.setBigUint64(20, offset, true)
   return makeUint8Array(header)
 }
+
+export function unicodePathExtraField(file: ZipFileDescription) {
+  if (!file.utf8) {
+    // make typescript compiler happy
+    throw new Error('file.utf8 must be set')
+  }
```

Comment:
You don't really need to do this if you know the utf8 property is always set when this function is called. Just add a `!` to let the TS compiler know. Same thing as with the `crc` and `uncompressedSize` properties.